project-fortis-spark/src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/cassandra/CassandraEventsSink.scala

```diff
@@ -75,63 +78,46 @@ object CassandraEventsSink extends Loggable {
     }
 
     def writeFortisEvents(events: RDD[Event]): Unit = {
-      val startTime = currentTimeMillis()
-      val success = Try(events.saveToCassandra(KeyspaceName, TableEvent, writeConf = WriteConf(ifNotExists = true))).isSuccess
-      FortisTelemetry.get.logDependency("sinks.cassandra", s"write.$TableEvent", success, currentTimeMillis() - startTime)
+      Timer.time(Log.logDependency("sinks.cassandra", s"write.$TableEvent", _, _)) {
+        events.saveToCassandra(KeyspaceName, TableEvent, writeConf = new WriteConf(ifNotExists = true))
+      }
     }
 
-    def removeDuplicates(batchid: String, events: RDD[Event]): RDD[Event] = {
-      val repartitionStartTime = currentTimeMillis()
-      val repartitionSuccess = Try(events.repartitionByCassandraReplica(KeyspaceName, TableEventBatches)).isSuccess
-      val repartitionEndTime = currentTimeMillis()
-      FortisTelemetry.get.logDependency("sinks.cassandra", s"repartition.$TableEventBatches", repartitionSuccess, repartitionEndTime - repartitionStartTime)
+    def withoutDuplicates(events: RDD[Event]): RDD[Event] = {
+      events.repartitionByCassandraReplica(KeyspaceName, TableEventBatches)
 
-      val joinStartTime = currentTimeMillis()
-      Try(events.joinWithCassandraTable(
+      events.joinWithCassandraTable(
         KeyspaceName,
         TableEventBatches,
         selectedColumns = SomeColumns("eventid", "batchid"),
         joinColumns = SomeColumns("eventid", "batchid")
-      ).map(_._1).cache()) match {
-        case Success(joined) =>
-          FortisTelemetry.get.logDependency("sinks.cassandra", s"join.$TableEventBatches", success = true, currentTimeMillis() - joinStartTime)
-          joined
-        case Failure(ex) =>
-          FortisTelemetry.get.logDependency("sinks.cassandra", s"join.$TableEventBatches", success = false, currentTimeMillis() - joinStartTime)
-          throw ex
-      }
+      ).map(_._1).cache()
     }
 
-    def writeEventBatchToEventTagTables(events: RDD[Event], sparkContext: SparkContext): Unit = {
-      val defaultZoom = configurationManager.fetchSiteSettings(sparkContext).defaultzoom
+    def writeEventPlaces(events: RDD[Event], sparkContext: SparkContext): Unit = {
+      Try {
+        Timer.time(Log.logDependency("sinks.cassandra", s"write.$TableEventPlaces", _, _)) {
```

Comment:
Previously we logged success/failure of the dependency based on the result of the Try. How is this handled in the current approach? We want to have a logDependency call even if the dependency failed so that we can reflect the failure in the telemetry.

Reply:
Check out the [implementation](https://github.com/CatalystCode/project-fortis/blob/master/project-fortis-spark/src/main/scala/com/microsoft/partnercatalyst/fortis/spark/logging/Timer.scala) of `Timer.time`, which handles this.

In simple terms, the logging statement will be invoked even if the body of the provided block throws. After calling the logging statement with the failure, `Timer.time` will re-throw. The outer `Try` here catches and swallows the re-throw since we'd still like to run downstream aggregations.

Reply:
Got it. Thanks for clarifying.